.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeZoneExtractor.cs

```diff
@@ -51,7 +51,35 @@ private IEnumerable<Token> MatchLocationTimes(string text)
 
             var timeMatch = config.LocationTimeSuffixRegex.Matches(text);
 
-            if (timeMatch.Count != 0)
+            // Before calling a Find() in location matcher, check if all the matched suffixes by
+            // LocationTimeSuffixRegex are already inside tokens extracted by TimeZone matcher.
+            // If so, don't call the Find() as they have been extracted by TimeZone matcher, otherwise, call it.
```

Comment:
Is it worth it saving only the Find time? It seems there's more benefit to try and improve the earlier step.

Reply:
After sync offline, there's not much difference between the two alternatives.